### DISCUSSION TITLE
Will this package release for other distros?

### DISCUSSION THREAD

**Discussion by ID86s** [JacKeTUs/universal-pidff#37] [General]
i can't find it for gentoo as emerge package, will it release for other packages managers?

**Comment by Lawstorant** [comment]
Only if someone using Gentoo will actually package it. This is a community-driven endeavor. I only use Arch so I'm maintaining an AUR package.

**Comment by r-burns** [comment]
Hi! Just wanted to say I'm happily using this on NixOS. Thanks for your work on this! I packaged it up like this:

<details>

```nix
{
  stdenv,
  lib,
  fetchFromGitHub,
  kernel,
}:

stdenv.mkDerivation {
  pname = "moza-ff";
  version = "unstable-2024-05-29";

  src = fetchFromGitHub {
    owner = "JacKeTUs";
    repo = "moza-ff";
    rev = "1ac6938ed4f553183950f12f89f6397f4cf4f605";
    hash = "sha256-XPg4VhYHFAUJF+zGO/Uj676Pjbp1L2N1rldxw8wLaAA=";
  };

  nativeBuildInputs = kernel.moduleBuildDependencies;

  postPatch = ''
    sed -i 's|depmod|#depmod|' Makefile
  '';

  makeFlags = kernel.makeFlags ++ [
    "KDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
  ];

  installFlags = [
    "INSTALL_MOD_PATH=${placeholder "out"}"
  ];

  meta = with lib; {
    description = "Force feedback support for MOZA steering wheel";
    homepage = "https://github.com/JacKeTUs/moza-ff";
    license = licenses.gpl2Plus;
    maintainers = [ maintainers.r-burns ];
    platforms = platforms.linux;
  };
}
```

</details>

I'm a nixpkgs committer and I'd be happy to contribute this and maintain it if any other NixOS users want to use the module.

**Comment by JacKeTUs** [comment]
Hello! Just to let you know: we think "universal-pidff" will better suite the purpose of bringing direct-drive wheels to the life on Linux
Please, update homepage/project name/repo URL to it. Thanks!

**Comment by DaRacci** [comment]
@r-burns I'm using NixOS too, if you want to make a PR to nixpkgs I'd be happy to be tagged as a maintainer for it too.
Would you by chance also maybe have a boxflat package working, I'm having some issues with it myself.

**Comment by Lawstorant** [comment]
@DaRacci I recommend just using the Flatpak version available on [Flathub](https://flathub.org/apps/io.github.lawstorant.boxflat). You only need to add one udev rule manually and you're good to go, without the need to package and maintain anything. I myself use it and only maintain the AUR package as it's trivial.

**Comment by computerdane** [comment]
@r-burns @DaRacci  I am also on NixOS and I packaged both universal-pidff and boxflat in [this repo](https://github.com/danerieber/moza-racing-wheel-nix)

**Comment by DaRacci** [comment]
> @DaRacci I recommend just using the Flatpak version available on [Flathub](https://flathub.org/apps/io.github.lawstorant.boxflat). You only need to add one udev rule manually and you're good to go, without the need to package and maintain anything. I myself use it and only maintain the AUR package as it's trivial.

While i could add Flatpak support for my System, I currently don't have it nor do i want to enable Flatpak, its just a matter of preference but i would much rather have a native build.

@danerieber If you are up for it, it may be worth upstreaming this into nixpkgs, if you end up doing this I'd be happy to help maintain it.

**Comment by computerdane** [comment]
> @danerieber If you are up for it, it may be worth upstreaming this into nixpkgs, if you end up doing this I'd be happy to help maintain it.

@DaRacci I made a fork with the package [here](https://github.com/danerieber/nixpkgs/tree/universal-pidff). I don't have enough time to properly test it right now, and I also need to sit down and read the PR guidelines for nixpkgs, but it would be awesome if you could test it on your hardware and lmk if it works!

**Comment by DaRacci** [comment]
@danerieber I don't actually have a moza wheel to test with yet, I've got one on a preorder and I've just been prepping for it, but the kernel compiles and loads with the module correctly, I've used nixpkgs-review to test it too.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>23 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_19.universal-pidff</li>
    <li>linuxKernel.packages.linux_4_19_hardened.universal-pidff</li>
    <li>linuxKernel.packages.linux_5_10.universal-pidff</li>
    <li>linuxKernel.packages.linux_5_10_hardened.universal-pidff</li>
    <li>linuxKernel.packages.linux_5_15.universal-pidff</li>
    <li>linuxKernel.packages.linux_5_15_hardened.universal-pidff</li>
    <li>linuxKernel.packages.linux_5_4.universal-pidff</li>
    <li>linuxKernel.packages.linux_5_4_hardened.universal-pidff</li>
    <li>linuxKernel.packages.linux_6_1.universal-pidff</li>
    <li>linuxKernel.packages.linux_6_10.universal-pidff</li>
    <li>linuxKernel.packages.linux_6_1_hardened.universal-pidff</li>
    <li>linuxKernel.packages.linux_6_6.universal-pidff</li>
    <li>linuxKernel.packages.linux_hardened.universal-pidff (linuxKernel.packages.linux_6_6_hardened.universal-pidff)</li>
    <li>linuxKernel.packages.linux_6_8.universal-pidff</li>
    <li>linuxKernel.packages.linux_6_8_hardened.universal-pidff</li>
    <li>linuxKernel.packages.linux_6_9.universal-pidff</li>
    <li>linuxKernel.packages.linux_6_9_hardened.universal-pidff</li>
    <li>linuxKernel.packages.linux_latest_libre.universal-pidff</li>
    <li>linuxKernel.packages.linux_libre.universal-pidff</li>
    <li>linuxKernel.packages.linux_lqx.universal-pidff</li>
    <li>linuxKernel.packages.linux_xanmod.universal-pidff</li>
    <li>linuxKernel.packages.linux_xanmod_latest.universal-pidff (linuxKernel.packages.linux_xanmod_stable.universal-pidff)</li>
    <li>linuxKernel.packages.linux_zen.universal-pidff</li>
  </ul>
</details>

**Comment by computerdane** [comment]
@DaRacci @r-burns I added you both as collaborators on my [nixpkgs fork](https://github.com/danerieber/nixpkgs/tree/universal-pidff). you can add yourselves as maintainers on the `universal-pidff` branch and then I'll open the PR!

**Comment by computerdane** [comment]
@DaRacci submitted! https://github.com/NixOS/nixpkgs/pull/334445

**Comment by JacKeTUs** [comment]
Hey guys, are we happy with NixOS packaging? Do we need to wait for someone implementing package for Gentoo emerge, or we can close this issue? :D

**Comment by DaRacci** [comment]
NixOS Side is happy, however it may be worth converting into a discussion for future Distro packaging talks.

**Comment by JPZ13** [comment]
@JacKeTUs - I'd like to package this repo as an RPM to be included in [akmods](https://github.com/ublue-os/akmods). Would you be open to me submitting a PR for the spec file?
